scipy/constants/_codata.py

```diff
@@ -1852,40 +1985,82 @@
 Wien wavelength displacement law constant                   2.897 771 955... e-3     (exact)                  m K
 W to Z mass ratio                                           0.881 45                 0.000 13                    """
 
-# -----------------------------------------------------------------------------
 
-physical_constants: dict[str, tuple[float, str, float]] = {}
+exact2022 = exact2018
+
 
+# -----------------------------------------------------------------------------
 
-def parse_constants_2002to2014(d: str) -> dict[str, tuple[float, str, float]]:
-    constants = {}
+
+def parse_constants_2002to2014(
+    d: str, exact_func: Callable[[Any], Any]
+) -> dict[str, tuple[float, str, float]]:
+    constants: dict[str, tuple[float, str, float]] = {}
+    exact: dict[str, float] = {}
+    need_replace = set()
     for line in d.split('\n'):
         name = line[:55].rstrip()
         val = float(line[55:77].replace(' ', '').replace('...', ''))
+        is_truncated = '...' in line[55:77]
+        is_exact = '(exact)' in line[77:99]
+        if is_truncated and is_exact:
+            # missing decimals, use computed exact value
+            need_replace.add(name)
+        elif is_exact:
+            exact[name] = val
```

Comment:
We need to collect all the exact, un-truncated values first; below, we go back and replace the truncated values.